src/p2p/net_node.inl

```diff
@@ -669,65 +742,68 @@ namespace nodetool
   template<class t_payload_net_handler>
   bool node_server<t_payload_net_handler>::store_config()
   {
-
     TRY_ENTRY();
-    if (!tools::create_directories_if_necessary(m_config_folder))
+
     {
-      MWARNING("Failed to create data directory: " << m_config_folder);
-      return false;
+      boost::system::error_code error{};
+      boost::filesystem::create_directories(m_config_folder, error);
+      if (error)
+      {
+        MWARNING("Failed to create data directory \"" << m_config_folder << "\": " << error.message());
+	return false;
+      }
```

Comment:
The function in epee isn't really needed because the call into boost is nearly identical anyway. Seemed like a waste of a function call. However, I reverted the change because it shouldn't be in this PR.